include/swift/AST/DiagnosticsSema.def

```diff
@@ -361,6 +361,10 @@ ERROR(conflicting_arguments_for_generic_parameter,none,
       "conflicting arguments to generic parameter %0 (%1)",
       (Type, StringRef))
 
+ERROR(ambiguous_arguments_for_generic_parameter,none,
+      "cannot disambiguate between types %1 and %2 for generic parameter "
+      "%0 in this context",
+      (Type, Type, Type))
```

Comment:
Does it make sense to offer a fix-it (it could insert a cast or add/remove @convention)?

Reply:
Hmmm ... I think it will be nice to suggest two possible fixes to use coercions (one for each type)... In this case, could be something like `insert as () -> Void` or `insert as @convention(block) ()-> Void` to disambiguate

Reply:
@theblixguy The problem with suggesting the fixes its is that we have to point the exact location because for disambiguating this we have to options:
```swift
foo(id(qux) as ()->Void)
// or 
foo(id(qux as @convention(block) ()-> Void))
```
And TBH I really don't know if we can get that right now ...